draft-ietf-acme-dns-account-01.mkd

```diff
@@ -62,44 +63,140 @@ systems or environments to handle challenge-solving for a single domain.
+that DNS name, and potentially its subdomains. This broad validation scope can
+have security implications, such as erroneously assuming control over the dns
+name and all of its subdomains.
+
+Furthermore, since each domain only has a single authorization label, it creates an impediment
+limiting the number of other entities domain validation can be delegated to.
+Delegating authorization to an entity requires the use of CNAME records, which can only used once
+per DNS name (or in this case, once per authorization label). This limitation requires that
+operators to pick a single ACME challenge solver for their domain name.
+
+In multi-region deployments, where separate availability zones
+serve the same content, and dependencies across them are avoided, operators need
+a way to obtain a separate certificate per zone, for the same domain name.
+Similarly, in cases of zero-downtime migration, two different setups of the
+infrastructure may coexist for a long period of time, and both need access to valid
 certificates.
 
-Due to the uniqueness of the `_acme-challenge` label, operators today have to
-pick a single ACME challenge solver for their domain name, and then add a
-`CNAME` record to this infrastructure. A domain name can only have one `CNAME`
-in DNS.
+This document specifies two new challenge types. `dns-02` and `dns-account-01`, which aim to address these deficiencies.
 
-This document specifies a new challenge type, `dns-account-01`. This challenge
-leverages the ACME Account Resource URL to present an account-unique stable
-challenge to an ACME server. This challenge allows any domain name to delegate
-its domain validation to more than one service through ACME account-unique DNS
-records.
-
-As now multiple labels can be used to prove domain control, and they depend on
-the ACME account, any number of them can be generated in advance, and then all
-required `CNAME` records can be created statically. The dynamic part of the
-label depends on the ACME account and not the account key, to allow for
-seamless account key rollover without the label changing. This ensures very
-long-lived labels, without any security considerations.
+This work follows all recommendations set forth in "Domain Control
+Validation using DNS" [I-D.draft-ietf-dnsop-domain-verification-techniques].
 
 This RFC does not intend to deprecate the `dns-01` challenge specified in
-{{!RFC8555}}. Since this new challenge does not modify or build on any
-pre-existing challenges, the ability to complete the `dns-account-01` challenge
-requires ACME server operators to deploy new changes to their codebase. This
-makes adopting and using this challenge an opt-in process.
+{{!RFC8555}}. Since these new challenges do not modify any pre-existing challenges,
+the ability to complete the `dns-02` or `dns-account-01` challenge requires ACME server
+operators to deploy new changes to their codebase. This makes adopting and using these
+challenges an opt-in process.
+
+## DNS-02
```

Comment:
There are headings
DNS-02
DNS-ACCOUNT-01
DNS-02 Challenge
....
DNS-ACCOUNT-01 Challenge

Is it possible to handle the full intro and challenge in the same section. It seems a bit awkward to me to go back and forth after just a few sentences of intro for each?

Reply:
I honestly don't know what the best way to format this. I suspect this is going to come up in the editorial review, so if you don't feel strongly about it, I rather just let it be decided there?

Reply:
OK, let's not block. I can try to make a pass after this lands and we can see what it looks like.